Soujanya2004/wanderlust-2024
# feat: 🎸 Add the Like listing feature

✅ **Closes**: #335

### Description
A clear and concise description of what the PR does.

This PR does the following:
- Adds a "like" feature to the listings on both the homepage and detail view.
- Updates the listing model to include a `likes` field to track the number of likes.
- Implements a controller function to handle the liking of listings.
- Modifies the views to display the like button and the current number of likes.

### Related Issues
Link any related issues using the format `Fixes #issue_number`.  
This helps to automatically close related issues when the PR is merged.

- Fixes #335 

### Changes
List of detailed changes made in this PR:
- Added a new feature to allow users to like listings.
- Refactored the listing model to include a `likes` field.
- Updated the listing controller to handle the like functionality.
- Modified the views (`index.ejs` and `show.ejs`) to include like buttons and display the number of likes.
- Updated the main application routes to include the new like functionality.

### Testing Instructions
Detailed instructions on how to test the changes. Include any setup needed and specific test cases.
1. Pull this branch.
2. Run `npm install` to install dependencies.
3. Run `npm test` to execute the test suite.
4. Verify that you can like a listing on both the homepage and detail view.
   - Check that the number of likes updates correctly after liking a listing.

### Screenshots (if applicable)
Add any screenshots that help explain or visualize the changes.
- [Insert screenshots here if applicable]

### Additional Context
Any additional context or information that reviewers should be aware of.

- This PR is based on the following user requirement to enhance user engagement by allowing them to like listings.

---

### Checklist
Make sure to check off all the items before submitting. Mark with `[x]` if done.


---

@Soujanya2004 Thank you for being patient, review this PR, and assign it to me with proper labels
